content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -149,6 +155,56 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+> Note that unlike `.dvc` files, `dvc.yaml` doesn't store any `md5` hash values,
+> making it extra easy to edit manually. These are kept in `dvc.lock` instead.
+
+### dvc.lock file
+
+The purpose of `dvc.lock` is to track the latest version of all files involved
+in the pipeline(s).
+
+Here's an example `dvc.lock` based on the one in [`dvc.yaml`](#dvcyaml-file)
+above:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - path: data/clean
+        md5: d8b874c5fa18c32b2d67f73606a1be60
+    params:
+      params.yaml:
+        levels.no: 5
+    outs:
+      - path: features
+        md5: 2119f7661d49546288b73b5730d76485
+    metrics:
+      - performance.json
+  ...
+    plots:
+      - logs.csv:
+          x: epoch
+          x_label: Epoch
+```
+
+It replicates the structure of `dvc.yaml`, but with some key differences and
```

Comment:
can we summarize the difference in some simpler way? w/o a long list that repeats everything else. What is the actual difference - only md5 and etag fields for outputs? anything else?

cc @skshetry

Reply:
I only listed the differences.

Reply:
Yes, for deps and outs, `md5/checksum/etag` and `path` are present in dvc.lock. For params, it's a key-value pair of {file, values}.

Reply:
I don't think, we need to explain the structure of `dvc.yaml` at all. No one needs to edit it, and the differences are clear from above example.

Reply:
Agree with both of you. That's what I have done.

> md5/checksum

Can you please clarify the field name @skshetry ? Using DVC 1.0 I've only seen `md5`. Is `checksum` the name of the field going forward or in certain cases? Thanks

Reply:
still too verbose to my mind @jorgeorpinel . I still don't think we need a list of params here.

Reply:
OK. I summarized it into a few short paragraphs instead of making another list (in b6d68dab). PTAL @shcheklein

Reply:
@jorgeorpinel, `checksum` was always there. It's used for `hdfs`. For http(s)/s3/azure/gs, we have `etag`, for `ssh/local` we have `md5` and for `hdfs`, we use `checksum` key.

Reply:
I've never seen `checksum` nor does it exist in docs 😱 and currently we only mention eTag for HTTP I think — will have to document all this... Created #1469

Reply:
And actually I addressed it for this PR's scope in f57f830e. Leaving this unresolved for future reference.